examples/purchase.py

```diff
@@ -0,0 +1,62 @@
+from collections.abc import Callable
+from dataclasses import dataclass
+from types import SimpleNamespace as State
+from typing import assert_type
+
+from app.entities import Customer
+from app.entities import CustomerId
+from app.entities import Order
+from app.entities import OrderId
+from app.entities import Payment
+from app.repositories import create_payment
+from app.repositories import load_customer
+from app.repositories import load_order
+from stories import I
+from stories import Story
+
+
+@dataclass
+class Purchase(Story):
+    I.find_order
+    I.find_customer
+    I.check_balance
+    I.persist_payment
+
+    class PurchaseState(State):
```

Comment:
This one should be untyped.

Typing examples should be in a separated chapters.